File: .claude/hooks/inject-datetime.mjs
#!/usr/bin/env node
// @hookstack user-prompt-inject-datetime
// Injecte la date et l'heure courantes dans chaque prompt (UserPromptSubmit)
import { fileURLToPath } from "node:url";

// Format neutre (ISO-like) : aucune locale codée en dur — déterministe pour les tests.
function pad(n) {
	return String(n).padStart(2, "0");
}

export function run({ now = new Date() } = {}) {
	const date = `${now.getFullYear()}-${pad(now.getMonth() + 1)}-${pad(now.getDate())}`;
	const time = `${pad(now.getHours())}:${pad(now.getMinutes())}`;
	const offsetMin = -now.getTimezoneOffset();
	const sign = offsetMin >= 0 ? "+" : "-";
	const tz = `UTC${sign}${pad(Math.floor(Math.abs(offsetMin) / 60))}:${pad(Math.abs(offsetMin) % 60)}`;
	return `Date et heure courantes : ${date} ${time} (${tz})\n`;
}

/* v8 ignore next 3 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	process.stdout.write(run());
}

File: .claude/hooks/inject-datetime.py
#!/usr/bin/env python3
# @hookstack user-prompt-inject-datetime
"""Injects the current date and time into every prompt (UserPromptSubmit)."""
import sys
from datetime import datetime, timedelta


def _format_neutral(dt):
    """Format ISO-like sans locale codée en dur — déterministe pour les tests."""
    offset = dt.utcoffset() or timedelta(0)
    total_min = int(offset.total_seconds() // 60)
    sign = "+" if total_min >= 0 else "-"
    tz = f"UTC{sign}{abs(total_min) // 60:02d}:{abs(total_min) % 60:02d}"
    return f"{dt:%Y-%m-%d %H:%M} ({tz})"


def run(*, now=None):
    if now is None:
        now = datetime.now().astimezone()
    return f"Date et heure courantes : {_format_neutral(now)}\n"


if __name__ == "__main__":
    sys.stdout.write(run())
